### ISSUE TITLE
HTTPClient.configuration should expose the configuration set

### ISSUE THREAD

**Issue by weissi** [swift-server/async-http-client#774]
For some reason, HTTPClient hides the configuration that was used to configure it from the user. We shouldn't do that.

**Comment by MartinLau7** [comment]
I agree with @weissi 's point of view. Since the configuration of `HTTPClient.shared` is currently hidden, it is impossible to modify certain properties of `configuration` when using `.shared`. Was this intentional by design? I want to restrict redirects for `HTTP status 302`, but this limitation forces me to abandon `.shared`, which is quite frustrating.

**Comment by MartinLau7** [comment]
In addition to allowing global configuration to be readable, I believe the global configuration for HTTPClient should not include redirectConfiguration and decompression. Alternatively, it should allow modifications to redirectConfiguration and decompression on a per-request basis.
